Mackbellemore/soen-390-team07
# feat(server): Production End Point

#

## Changes this PR introduces

- Production endpoints to create both bike and part
- Initialize part with a cost in development (Init.js file)
<del>- Associate price with an entity created, the selling price is 30% higher than the cost price </del>
- Allow user to set profit margin



## How to test this PR

- Test each end point, create a production-part as well as a production-bike. 
- Check the price associated with that part or bike
- Make sure that this part or bike is also created

```
{
    "type": "Part",
    "componentDetail": {
        "name": "handle_bar4222256222",
        "description": "great handle bar",
        "quality": "alluminium",
        "color": "red",
        "type": "handle_bar",
        "detail": "thicc",
        "profitMargin" : "2"
    },
    "status": "Ongoing",
    "quantity": 2,
    "startDate": "2021-01-01",
    "endDate": "2021-01-03",
    "assemblyMachine": "R2-D2",
    "note": "Parts..."
}


{
    "type": "Bike",
    "componentDetail": {
        "name": "carbon road bike",
        "description": "some_description",
        "weightAmount": 5,
        "weightType": "kg",
        "color": "blue",
        "parts": {
            "handle_bar": "606917576e554255f8b0fce9",
            "wheels" : "606917576e554255f8b0fce3",
            "chain" : "606917576e554255f8b0fce4",
            "frame" : "606917576e554255f8b0fce5",
            "pedal" : "606917576e554255f8b0fce6",
            "brakes": "606917576e554255f8b0fce7",
            "seat": "606917576e554255f8b0fce8",
            "fork": "606917576e554255f8b0fce9"
        },
        "profitMargin" : "3"
    },
    "status": "Ongoing",
    "quantity": 2,
    "startDate": "2021-01-01",
    "endDate": "2021-01-03",
    "assemblyMachine": "C-3P0",
    "note": "Bikes..."
}
```







## Notes for reviewers

- Be as critical as possible 💀



## Checklist
